### PR TITLE
[REVIEW] cuSpatial trajectory clustering notebook

### PR DESCRIPTION
**Pull request by harrism**
This PR:

- [x] adds a new folder for cuSpatial notebooks (`cuspatial`). 
- [x] Adds a new cuSpatial trajectory clustering example notebook and data
- [x] Updates the Readme to link to this notebook.

### CODE REVIEW

File: cuspatial/trajectory_clustering.ipynb [hunk 1]
```diff
@@ -0,0 +1,720 @@
...
+    "\n",
+    "def hausforff_distance(u,v):\n",
+    "    return max(directed_hausdorff(u, v)[0], directed_hausdorff(v, u)[0])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load and prepare the data"
```
Comment (randerzander) [inline cuspatial/trajectory_clustering.ipynb]:
```suggestion
    "## Load and prepare the data"
    
    For comparison's sake, we'll first implement the CPU-only version of this analysis.
```

Perhaps you want to use different verbiage, but I had to scroll up and down a few times to confirm that the first several cells are host-only code.

It helps to describe step by step what's about to happen before providing the code.

File: cuspatial/trajectory_clustering.ipynb [hunk 2]
```diff
@@ -0,0 +1,720 @@
...
...p+Vs/8VWOj4/46FsfkOiMv+74G1G0XWdDXEvppVDk6QSjUlxwON/vA4Kv/611loNlTlYYrMto/ZQnV+d7Wth617BUOZ2PU7skVZg0Uu2GPtLVlJIsD0v63hECtJ3l4iIGZu+qjvPzbcyRsoJq60ayqCXJDUniaeuKfJKjk4IUjdTQNyO5LUTccmg9bnDjA06MRSmo0afH+FC4llCKMSQ4yRIm80mUVfoYT9C2A26Im6rdeosyBduqR4iA94LeRoT9wcGUrCgiMCXto39Ag3U9WZmxfbFmeny0z1kZrENKwdGdA9SIsO2aDg4CwfcgQWvJyxfVqLcOCOJmO0JcPIezkn6IQcBNFRfdtmmAgA8wW0z5+Aefcef+McUsY72uyIsUIQIiDKjxppQmmn1fvKg5Oiqp1z1CajabjpdXGw4XM3ZtG8flYsxAcTE0VshIMgzBkRclJnHkRSxInI2J90IahsHx9MlL3GAwRtNvo/9FCEWWC5ptS1ImvPHwLm58UCsRMFqQ5pLdqsePm4EiTyA4Fssl220VJ8Iu0NtreUD0na1OzymnBc2uQWuNMRmDs8TedECoKAFUeFwf2Gxi0ZckCfMHx5GOaj2XZzVCepyNmwZn4wSiaR1dFzAmp283tHWMdUgSkDqgbKDa9hgtEcLtPZpCCJT0lFODwpMXJSEEqm2DThJ6NyCl4uLliuXhfMz68+Sl5ulnFfffusvR0Zyr8zVaa7q2px8sgxDYXjD0gn5wpEm8x7TR9F1N07Rx415KghLx9yARHoKCoXGoIBEBVGGwXZxETcucprMspmX0SfQOJxyJ0FxjSQdnmRaKPMuxbsB5F6cizuNHo69W8d8JGaM03OBRWqB0xsVpQ6oEXedY3J+RK0WaKiZpXEyNjnIQ68d7G5hOZxwf34mxJcHFSAMXg7nDuOn2fiDJ0ij7kTJ2LFXMHBsGh/cSrwRaxvVC6ygLVFITRPy7rCjoe4dJSq52DUZrPJJdVYMb+OTRI3wI/Pf/9J/Sti0ff/wD7t27R9/3/M5v/za/9/V/y9tvv0nd9lycX1DVHd1g8SFQFAl21yBELJzc0MTNoUwRSlIYsG7gybPnlGVCVTX0vUXpCVma0nWWREY5ZPA/tIlvA94IOgeJCSS5BhlzGk1qECpKk72MzSmjS7IkxXu395DUVYtJDEoIpmWURl93Zu8dzPfxKJH
```
Comment (randerzander) [inline cuspatial/trajectory_clustering.ipynb]:
```suggestion
    "Now we'll calculate the `Hausdorff distance` matrix using the cuSpatial library (GPU). Keep an eye on the speed! :)"
```

File: cuspatial/trajectory_clustering.ipynb (hunk 1 above)
Comment (randerzander) [reply cuspatial/trajectory_clustering.ipynb]: A numbered list of what this notebook will do helps users follow along. Something like..

"We'll compare CPU only vs GPU accelerated implementations.

For the CPU version, we need to:
1. Load the data from disk
2. Define trajectory calculation functions
3. Filter out noise and apply the trajectory calculations
4. Do a visual inspection of the computed trajectories
5. Cluster the trajectories using Hausdorff distance as the similarity metric

...
much code
...

For the GPU accelerated implementation, we implement the same logic with mostly builtin functions:
1. foo
2. bar
3. profit

File: cuspatial/trajectory_clustering.ipynb [hunk 3]
```diff
@@ -0,0 +1,720 @@
...
+    "\n",
+    "We use Hausdorff distance as the similarity metric between trajectories. We compute the pairwise similarity matrix and apply AgglomerativeClustering (AC) and DBSCAN afterwards.\n",
+    "\n",
+    "In this example, cuSpatial can significantly accelerate the computation of the Hausdorff distance similarity matrix. Comparing to the typical scikit-learn implementation of Hausdorff distance (single CPU thread), cuSpatial reduces the computation time from about 5 hours to under 15 seconds on this particular dataset.\n",
+    "\n",
+    "Since we pre-computed the similarity matrix, we can much more easily experiment with different clustering algorithm and different hyperparameter sets. In this example we do it in an interactive way using ipython widgets, you may play with the hyperparameter set and see how the clustering result responds.\n",
+    "\n",
+    "## Install Dependencies\n",
+    "\t\n",
+    "1. cuSpatial:    `conda install -c rapidsai-nightly cuspatial`\n",
```
[the remark this answers is not in the document]
Comment (harrism) [reply cuspatial/trajectory_clustering.ipynb]: I got this install line from Keith. Since I build from source, I've never tried it.

**Pull request merged**